_pages/documentation/gem5-stdlib/7-using-local-data-sources.md

```diff
@@ -0,0 +1,168 @@
+
+#### Creating a Custom Resource Data Source
+
+Let's begin by downloading the resource we'll use as our local source. I'm selecting the `riscv-hello` resource from gem5 resources.
+
+```bash
+wget https://storage.googleapis.com/dist.gem5.org/dist/develop/test-progs/hello/bin/riscv/linux/hello-20220728
+```
+
+Next, let's create the JSON data source. I'll name the file `my-resources.json`. The contents should look like this:
+
+```json
+[
+    {
+        "category": "binary",
+        "id": "riscv-hello",
+        "description": "A 'Hello World!' binary, compiled to RISCV.",
+        "architecture": "RISCV",
+        "size": 4674040,
+        "tags": [
+            "hello"
+        ],
+        "is_zipped": false,
+        "md5sum": "6d9494d22b90d817e826b0d762fda973",
+        "source": "src/simple",
+        "url": "file://<path/to/the/above/downloaded/file>",
+        "code_examples": [
+            {
+                "example": "https://github.com/gem5/gem5/tree/develop/configs/example/gem5_library/checkpoints/riscv-hello-restore-checkpoint.py",
+                "tested": true
+            },
+            {
+                "example": "https://github.com/gem5/gem5/tree/develop/configs/example/gem5_library/checkpoints/riscv-hello-save-checkpoint.py",
+                "tested": true
+            },
+            {
+                "example": "https://github.com/gem5/gem5/tree/develop/configs/example/gem5_library/riscvmatched-hello.py",
+                "tested": true
+            }
+        ],
+        "license": "",
+        "author": [],
+        "source_url": "https://github.com/gem5/gem5-resources/tree/develop/src/simple",
+        "resource_version": "1.0.0",
+        "gem5_versions": [
+            "23.0"
+        ],
+        "example_usage": "obtain_resource(resource_id=\"riscv-hello\")"
+    }
+]
+```
+
+The JSON object of a resource should adhere to the [gem5 resources schema](https://resources.gem5.org/gem5-resources-schema.json). 
+
+**Note**: While the `url` field can be a link, in this case, I'm using a local download.
+
+#### Creating Your Custom Resource Configuration
+
+Create a file named `gem5-config.json` with the following content:
+
+```json
+{
+    "sources": {
+        "my-json-data-source": {
+            "url": "path/to/my-resources.json",
+            "isMongo": false
+        }
+    }
+}
+```
+
+#### Running gem5 with a Local Data Source
+
+First, build gem5 with RISC-V:
+
+```bash
+scons build/RISCV/gem5.opt -j`nproc`
+```
+
+Next, run the `riscvmatched-hello.py` example using our local `riscv-hello` resource:
+
+```bash
+GEM5_CONFIG=path/to/gem5-config.json ./build/RISCV/gem5.opt configs/example/gem5_library/riscvmatched-hello.py
+```
+
+This command will execute the `riscvmatched-hello.py` script using our locally downloaded resource.
```

Comment:
I'd create something more fake here. Just have them write a config that calls `obtain_resource` then print off some data from the resource like the local path. Then mention this returned resource can be used in a simulation, just like those obtained from the official gem5 MongoDB datasource.